### PR TITLE
Adding docs for pulumi-terraform Go and DotNet SDKs

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #3336

### CODE REVIEW

File: content/docs/guides/adopting/from_terraform.md [hunk 1]
```diff
@@ -144,18 +148,92 @@ for i in range(2):
...
+        publicSubnetIds := stateRef.Outputs.ApplyT(func(args interface{}) ([]string, error) {
+            ids := args.(map[string]interface{})["public_subnet_ids"].([]interface{})
+            subnetIds := make([]string, len(ids))
+            for i, v := range ids {
+                subnetIds[i] = v.(string)
+            }
+            return subnetIds, nil
+        }).(pulumi.StringArrayOutput)
+        for x := 0; x < 2; x++ {
+            ii := x
```
Comment (lukehoban) [inline content/docs/guides/adopting/from_terraform.md:183]: This `ii` is not required given how this code now reads.

File: content/docs/guides/adopting/from_terraform.md [hunk 2]
```diff
@@ -144,18 +148,92 @@ for i in range(2):
...
+            for (int i = 0; i < 2; i++)
+            {
+                var server = new Instance($"instance-{i}", new InstanceArgs
+                {
+                    Ami = "ami-7172b611",
+                    InstanceType = "t2.micro",
+                    SubnetId = subnetIds.GetAt(i),
+                });
+            }
+        });
```
Comment (lukehoban) [inline content/docs/guides/adopting/from_terraform.md:234]: I thought we now encouraged a style in C# that uses a class inheriting from `Stack` - like https://github.com/pulumi/templates/blob/master/azure-csharp/MyStack.cs#L5.

File: content/docs/guides/adopting/from_terraform.md [hunk 3]
```diff
@@ -252,18 +330,70 @@ network_state = terraform.state.RemoteStateReference('network',
...
+            Token:        pulumi.StringPtrInput(token.(pulumi.StringOutput)),
+            Workspaces: state.WorkspaceStateArgs{
+                Name: pulumi.String(workspace),
+            },
+        })
+        if err != nil {
+            return err
+        }
+
+        ctx.Export("test", state.Outputs)
```
Comment (lukehoban) [inline content/docs/guides/adopting/from_terraform.md:361]: I don't see the equivalent of this in the other examples?

File: content/docs/guides/adopting/from_terraform.md [hunk 4]
```diff
@@ -252,18 +330,70 @@ network_state = terraform.state.RemoteStateReference('network',
...
+	pulumi.Run(func(ctx *pulumi.Context) error {
+
+        conf := config.New(ctx, "")
+        token := conf.RequireSecret("tfeToken")
+        organization := conf.Require("organization")
+        workspace := conf.Require("workspaceName")
+
+        state, err := state.NewRemoteStateReference(ctx, "remote-backend-state", &state.RemoteBackendStateArgs{
+            Organization: pulumi.String(organization),
+            Token:        pulumi.StringPtrInput(token.(pulumi.StringOutput)),
```
Comment (lukehoban) [inline content/docs/guides/adopting/from_terraform.md:352]: Doesn't this work?

```suggestion
            Token:        token.(pulumi.StringOutput),
```

FWIW - this *should* be able to be just `token`, but I think there is a bug in the typing of `conf.*Secret`.

Comment (lukehoban) [reply content/docs/guides/adopting/from_terraform.md:352]: > FWIW - this should be able to be just token, but I think there is a bug in the typing of conf.*Secret.

https://github.com/pulumi/pulumi/issues/4664

Comment (stack72) [reply content/docs/guides/adopting/from_terraform.md]: You are correct :)

File: content/docs/guides/adopting/from_terraform.md (hunk 3 above)
Comment (stack72) [reply content/docs/guides/adopting/from_terraform.md]: gone!

File: content/docs/guides/adopting/from_terraform.md (hunk 2 above)
Comment (stack72) [reply content/docs/guides/adopting/from_terraform.md]: updated

**Pull request merged**